build/cmake/ResSearch.cmake

```diff
@@ -0,0 +1,57 @@
+include(CheckSymbolExists)
+include(CMakePushCheckState)
+
+cmake_push_check_state(RESET)
+
+# The name of the library that performs name resolution, suitable for giving to the "-l" link flag
+set(RESOLVE_LIB_NAME)
+# If TRUE, then the C runtime provides the name resolution that we need
+set(resolve_is_libc FALSE)
+
+if(WIN32)
+    set(RESOLVE_LIB_NAME Dnsapi)
+    set(_MONGOC_HAVE_DNSAPI 1)
+else()
+    # Try to find the search functions for various configurations.
+    # Headers required by minimum on the strictest system: (Tested on FreeBSD 13)
+    set(resolve_headers netinet/in.h sys/types.h arpa/nameser.h resolv.h)
+    set(CMAKE_REQUIRED_LIBRARIES resolv)
+    check_symbol_exists(res_nsearch "${resolve_headers}" _MONGOC_HAVE_RES_NSEARCH_RESOLV)
+    check_symbol_exists(res_search "${resolve_headers}" _MONGOC_HAVE_RES_SEARCH_RESOLV)
+    check_symbol_exists(res_ndestroy "${resolve_headers}" _MONGOC_HAVE_RES_NDESTROY_RESOLV)
+    check_symbol_exists(res_nclose "${resolve_headers}" _MONGOC_HAVE_RES_NCLOSE_RESOLV)
+    if((_MONGOC_HAVE_RES_NSEARCH_RESOLV OR _MONGOC_HAVE_RES_SEARCH_RESOLV)
+        AND (_MONGOC_HAVE_RES_NDESTROY_RESOLV OR _MONGOC_HAVE_RES_NCLOSE_RESOLV))
+        set(RESOLVE_LIB_NAME resolv)
+    else()
+        # Can we use name resolution with just libc?
+        unset(CMAKE_REQUIRED_LIBRARIES)
+        check_symbol_exists(res_nsearch "${resolve_headers}" _MONGOC_HAVE_RES_NSEARCH_NOLINK)
+        check_symbol_exists(res_search "${resolve_headers}" _MONGOC_HAVE_RES_SEARCH_NOLINK)
+        check_symbol_exists(res_ndestroy "${resolve_headers}" _MONGOC_HAVE_RES_NDESTROY_NOLINK)
+        check_symbol_exists(res_nclose "${resolve_headers}" _MONGOC_HAVE_RES_NCLOSE_NOLINK)
```

Comment:
It cannot, since `check_symbol_exists` caches based only on the name of the variable, not any of the attributes being checked, so the second check would always be skipped.

Reply:
Ah, I missed the "will be created as an internal cache variable" part of the specification. Good catch.